### PR TITLE
Fix bug in leaf importance calculation

### PR DESCRIPTION
**Pull request by maxhutch**
The local variance computation, which is used to weight the importance,
has a weights-related bug that results in negative importance.

CC: @jling11

### CODE REVIEW

File: src/test/scala/io/citrine/lolo/trees/regression/RegressionTreeTest.scala [hunk 1]
```diff
@@ -187,6 +207,6 @@ object RegressionTreeTest {
     new RegressionTreeTest().longerTest()
     new RegressionTreeTest().testCategorical()
     new RegressionTreeTest().testLinearLeaves()
-    new RegressionTreeTest().testShortTreeWithLinearLeaf()
```
Comment (jling11) [inline src/test/scala/io/citrine/lolo/trees/regression/RegressionTreeTest.scala]: Don't you want to add the testStumpwithLinearLeaves test ?

File: src/test/scala/io/citrine/lolo/trees/regression/RegressionTreeTest.scala [hunk 2]
```diff
@@ -173,6 +175,24 @@ class RegressionTreeTest {
     /* They should have roughly the same value (linear regression has some noise in it) */
     assert(Math.abs(importances(0) - 0.30030792928576033) < 1.0e-3)
   }
+
+  /**
+    * Test with random weights
```
Comment (jling11) [inline src/test/scala/io/citrine/lolo/trees/regression/RegressionTreeTest.scala]: "Test with random weights to ensure all feature importances of stump tree with linear leaves are non-negative."

**Comment by sparadiso** [comment]
LGTM - @jling11 ?

File: src/main/scala/io/citrine/lolo/trees/regression/RegressionTrainingLeaf.scala [hunk 3]
```diff
@@ -31,10 +31,10 @@ class RegressionTrainingLeaf(
   def getFeatureImportance(): scala.collection.mutable.ArraySeq[Double] = {
```
Comment (jling11) [inline src/main/scala/io/citrine/lolo/trees/regression/RegressionTrainingLeaf.scala]: I would add on to this comment with the formula used to calculate the impurity now:
sum(L**2 w)/ sum(w) - (sum(L*w))**2/w**2, where L is the regression label, W is the weight.

File: src/test/scala/io/citrine/lolo/trees/regression/RegressionTreeTest.scala (hunk 1 above)
Comment (maxhutch) [inline src/test/scala/io/citrine/lolo/trees/regression/RegressionTreeTest.scala]: This is just a driver for human use.  The tests are run whenever they have `@Test` annotations

**Pull request merged**